railties/lib/rails/test_help.rb

```diff
@@ -31,6 +31,10 @@
   ActiveSupport.on_load(:action_dispatch_integration_test) do
     self.fixture_path = ActiveSupport::TestCase.fixture_path
   end
+else
+  ActiveSupport.on_load(:active_support_test_case) do
```

Comment:
It seems like the `if defined?(ActiveRecord::Base)` check should only wrap this code:

```ruby
  begin
    ActiveRecord::Migration.maintain_test_schema!
  rescue ActiveRecord::PendingMigrationError => e
    puts e.to_s.strip
    exit 1
  end
```

Since nothing else inside that if statement interacts with Active Record.